### PR TITLE
Direct livestreaming improvements

### PR DESCRIPTION
**Pull request by ktprograms**
### Use selectedChannel instead of first in list when livestreaming

Don't use `channels[0]` anywhere since that makes selecting a channel useless.

### Wait for stream claim to be confirmed before starting

Allowing the stream to be started before it is confirmed could cause viewers to miss the start of the producer's stream (in the time that it takes to confirm).

### CODE REVIEW

File: Odysee/Controllers/Library/GoLiveViewController.swift [hunk 1]
```diff
@@ -485,8 +511,14 @@ class GoLiveViewController: UIViewController, UIPickerViewDataSource, UIPickerVi
                     return
                 }
 
-                // The claim was successfully set up. Create the stream key and start
-                self.signAndSetupStream(channel: self.channels[0])
+                // The claim was successfully set up.
+                // Wait for the claim to be confirmed, then create the stream key and start
+                let result = data["result"] as? [String: Any]
+                guard let txid = result?["txid"] as? String else {
+                    self.displayRequirementNotMet(message: String.localized("Could not get txid from publish API call."))
```
Comment (ktprograms) [inline Odysee/Controllers/Library/GoLiveViewController.swift:518]: Is this error message okay?

Comment (akinwale) [reply Odysee/Controllers/Library/GoLiveViewController.swift:518]: We can use a more user-friendly error message: "The livestream could not be created at the moment. Please try again later."

**Review by akinwale — CHANGES_REQUESTED** [review]
Just one small change. Everything else looks good. Thanks.

File: Odysee/Controllers/Library/GoLiveViewController.swift (hunk 1 above)
Comment (ktprograms) [reply Odysee/Controllers/Library/GoLiveViewController.swift:518]: What if I put the user friendly message there, but added a `showError` call with this dev-facing message for easier debugging if it's ever needed in future?

**Pull request merged**